standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStore.java

```diff
@@ -3089,7 +3089,7 @@ public Table get_table(final String dbname, final String name) throws MetaExcept
         NoSuchObjectException {
       String[] parsedDbName = parseDbName(dbname, conf);
       return getTableInternal(
-            parsedDbName[CAT_NAME], parsedDbName[DB_NAME], name, null, null, false);
+            parsedDbName[CAT_NAME], parsedDbName[DB_NAME], name, null, null, false, null);
```

Comment:
Observe in this case, retrieve column stats (parameter before null) is false. Basically, when we retrieve a given table, if we want to retrieve stats, we will only retrieve stats for a given engine. Since this is not a public API but rather a private method, I am thinking whether I should eliminate the boolean and document behavior as 'if engine is specified, retrieve column stats for that engine; if engine is null, do not retrieve column stats'. What do you think?

Reply:
Started making this change and there are already some boolean in the metastore thrift API that probably we would need to deprecate. To not make this patch any more complex, I just documented the API properly and added a precondition at the start of the method.